src/org/rascalmpl/library/Location.rsc

```diff
@@ -43,6 +43,63 @@ loc relativize(list[loc] haystack, loc needle) {
     }
 }
 
+@synopsis{Names of file system path syntaxes that can be found in the wild.}
+@description{
+* `generic` captures all unix-like systems like the BSD family, SunOs, Solaris, Irix, etc.
+* `mac_osx` captures modern Mac's that also have a unix-like filesystem but with different exceptions to permissable characters and case-sensitivity.
+* `windows` is for all DOS-based filesystems and what came after, with the slashes in the other direction.
+* `linux` is for all the filesystems in Linux distributions.
+
+Use this as a parameter to ((locFromFileSystem)) to choose how to parse a string as a filesystem path.
+}
+data FileSystemSyntax
+    = generic()
+    | mac_osx()
+    | windows()
+    | linux()
+    ;
+
+@javaClass{org.rascalmpl.library.Prelude}
+@synopsis{Utility to retrieve the current notation for file system paths.}
+@description{
+This is uses to configure the default syntax parameter of ((locFromFileSystem)).
+}
+java FileSystemSyntax currentFileSystem();
+
+@synopsis{Converts the OS-specific string representation of a file or directory PATH to a `loc`.}
+@description{
+This converts a string that hold a path to a file, written in a specific notation for paths, to
+a canonical `loc` in URI format.
+
+* if `legalize` is true, then every path segment is legalized by replacing offending characters to `_`
+* if `legalize` is false, and offending characters appear between the path separators, then an IO exception is thrown.
+* if the requested file system syntax is either case insensitive or not case preserving (or both), then all uppercase characters will be replaced by lowercase characters.
+* on windows systems the drive letter `C:` is added if a drive letter is missing.
+* on all the other systems, if the path starts with a path separator, it is taken as absolute in `file:///`. Otherwise the root will be `cwd:///`.
```

Comment:
Yes we could also support that, however this library from Apache does not and UNC has many particularities that make it quite a different story to cover to URI. I believe we should reuse a library for that or reuse java `new File` or `new URL` that does support UNC. However,  these two are extremely slow because they insist on setting up a network connection. All we want here is a parser and an isomorphic translation... 

The jury is still out. But I'm leaning towards a full syntax definition with a concrete tree processor for the mapping.